### PR TITLE
Unify use of vmi_lock_guard in libusermode

### PR DESCRIPTION
**Pull request by disaykin**
Unify the use of vmi_lock_guard in libusermode

### CODE REVIEW

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

File: src/plugins/memdump/userhook.cpp [hunk 1]
```diff
@@ -219,13 +219,8 @@ void memdump::setup_dotnet_hooks(drakvuf_t drakvuf, const char* dll_name, const
         return;
     }
 
-    plugin_target_config_entry_t entry;
-    entry.function_name = "AssemblyNative::LoadImage";
-    entry.dll_name = dll_name;
-    entry.type = HOOK_BY_OFFSET;
-    entry.offset = func_rva;
-    entry.actions = HookActions::empty().set_log().set_stack();
-    this->wanted_hooks.add_hook(std::move(entry));
```
Comment (tklengyel) [inline src/plugins/memdump/userhook.cpp:228]: This seems to be an unrelated change

Comment (disaykin) [reply src/plugins/memdump/userhook.cpp:228]: Right. I decided not to move this change in a separate request. But I can do if you ask.

Comment (disaykin) [reply src/plugins/memdump/userhook.cpp]: I remove this change from PR
